src/dmd/expressionsem.d

```diff
@@ -5130,12 +5130,15 @@ private extern (C++) final class ExpressionSemanticVisitor : Visitor
 
     override void visit(DeleteExp exp)
     {
-        // @@@DEPRECATED_2019-02@@@
-        // 1. Deprecation for 1 year
-        // 2. Error for 1 year
-        // 3. Removal of keyword, "delete" can be used for other identities
-        if (!exp.isRAII)
-            deprecation(exp.loc, "The `delete` keyword has been deprecated.  Use `object.destroy()` (and `core.memory.GC.free()` if applicable) instead.");
+        if (!sc.isDeprecated)
+        {
+            // @@@DEPRECATED_2019-02@@@
+            // 1. Deprecation for 1 year
+            // 2. Error for 1 year
+            // 3. Removal of keyword, "delete" can be used for other identities
+            if (!exp.isRAII)
+                deprecation(exp.loc, "The `delete` keyword has been deprecated.  Use `object.destroy()` (and `core.memory.GC.free()` if applicable) instead.");
+        }
```

Comment:
The bug report was more general (before you changed the title).
I took delete as an example, because it's what hits me, but I'm pretty sure it affects other deprecations.
For example:
```D
deprecated void main ()
{
    int i, o;
    switch (i)
    {
    case 1:
        o = 1;
    case 2:
        o = 2;
        break;
    default:
    }
}
```

Reply:
OKay I created a new subissue 18647 of 17906 which just addresses the delete deprecation.
Maybe `deprecation` should require the current scope, s.t. `sc.isDeprecated` can automatically be called (and we don't need to patch this in at all places)?